thesis-de/content-de.tex

```diff
@@ -360,6 +362,128 @@ \subsubsection{Durchführung der Evaluierung}
 \end{outline}
 \nl%
 
+\subsection{Schnittstelle - Datenbereitstellung}
+\label{subsec:schnittstelle_datenbereitstellung}
+Neben der herkömmlichen Methode zur Datenabfrage von Datenbank-Systemen über
+einen sogenannten Datenbanktreiber, existiert noch eine weitere Möglichkeit in
+Form von einer sogenannten ``API''. API bedeuter im ausgeschriebenen
+``application programming interface'' und ermöglicht eine standardisierte
+Anbindung von externen Softwaresystemen. Dabei ist es egal, in welcher
+Programmiersprache oder auf welchem Protokoll das zu anbindende System arbeitet.
+
+Die Verwendung von einer Schnittstelle dieser Art besitzt Nachteile, sowie
+Vorteile beziehungsweise verbirgt gewisse Gefahren in Hinsicht auf den späteren
+Einsatz im Produktivsystem. Eine dieser Gefahren ist das
+vernachlässigen der Dokumentation einer API. Dadurch, das während der
+Entwicklung einer API bereits definiert wird, welche Fähigkeiten und Methodiken
+die API besitzen soll, wäre eine nicht vorhandene oder unvollständige
+Dokumentation suboptimal. Darum sollte bei der Auswahl von dem genutzen API
+System darauf geachtet werden, dass die Dokumentation von Softwarekomponenten
+für den Entwickler so einfach wie möglich erfolgt.
+
+Wenn die Dokumentation einer API vollständig und das ausführende System
+den Performance-Anforderungen gerecht ist, bringt die Nutzung von einem
+API-System gewisse Vorteile. Darunter ist ein besseres Management von
+Zugriffsrechten und somit auch gleichzeitig eine Verbesserung der
+Systemsicherheit. Durch den vorab festgelegten Funktionsumfang der
+API kann der Systeminhaber geziehlt kontrollieren, welche Inhalte ein Benutzer
+sehen oder modifizieren kann. Dabei können grobe Einschränkungen, wie das Lesen
+oder Schreiben auf der gesamten Datenbank, oder auch detaillierte Einschränkung,
+wie das Lesen oder Schreiben von einer einzelnen Tabelle in der Datenbank,
+kontrolliert werden. Anders wie bei der Verbindung über einen Datenbanktreiber
+kann bei der API der potentielle Angreifer keine eigenen Datenbankbefehle
+definieren und ausführen. Somit wird das Risiko der Datenmanipulation durch eine
```

Comment:
hinweis über transformation in 869d1847d259a974d37241c754aedf5296c780a7 eingefügt